### PR TITLE
914 - Rework the theme building setup

### LINKED ISSUE #425: Generate themes as part of CI and do not commit those files
This improves the DX because changes to themes won't be part of commits and PRs. At the moment, some PRs contain more changes to themes than actual component changes.

- [ ] Remove generated theme files from git
- [ ] Add `build:themes` as part of the release build
- [ ] Add `build:themes` as part of the local `yarn start --watch` pipeline so you get to see changes to tokens reflected in the themes in Storybook
- [ ] Make sure the theme NPM packages contain the generated files

### LINKED ISSUE #914: Fix NPM theme packages
- Even if a theme does not have a dark mode, we still generate `all.css` with media queries for light & dark mode. Using `all.css` will break your web app when the OS is in dark mode
- We name files `base.css` and `base.scss` which can be difficult with tooling. Some tooling may always prefer the `.scss` file extension even though the developer wants it to include the `.css` extension
- The files contain the tokens for *all* components; if you are only using 1 component, then a lot is unnecessary bloat. 

Solutions:
1. Check if there is a dark mode and if not, output the light tokens *without* a media query in `all.css`
2. Put the `*.scss` files in a `mixins` folder so there can be no confusion.
3. Consider splitting the tokens into different files (1 for each component?)

### PR DESCRIPTION
**Pull request by jpzwarte**
- Adjust the contents of the theme packages based on feedback from CFAs
- Only output light/dark files if the mode actually exists in the tokens
- Improve performance of building the themes (takes several minutes atm)
- Replace our own custom code with `style-dictionary` & `@tokens-studio/sd-transforms` code
- Remove generated theme files from git (fixes merge conflicts)
- Remove automated icon import from Figma and move to a manual `yarn import-icons` command

Fixes #914 
Fixes #425

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -207,22 +151,106 @@
...
         "lint:components:styles"
       ]
     },
     "lint:components:code": {
-      "command": "eslint --config packages/components/eslint.config.mjs packages/components/*"
+      "command": "eslint --config packages/components/eslint.config.mjs \"packages/components/**/*.ts\""
     },
     "lint:components:styles": {
-      "command": "stylelint --quiet-deprecation-warnings packages/components/*"
+      "command": "stylelint \"packages/components/**/*.{css,scss}\""
```
Comment (anna-lach) [inline package.json:248]:
```suggestion
      "command": "stylelint 'packages/components/**/*.{css,scss}'"
```

File: package.json [hunk 2]
```diff
@@ -207,22 +151,106 @@
...
+    },
+    "lint": {
       "dependencies": [
         "lint:components:code",
         "lint:components:styles"
       ]
     },
     "lint:components:code": {
-      "command": "eslint --config packages/components/eslint.config.mjs packages/components/*"
+      "command": "eslint --config packages/components/eslint.config.mjs \"packages/components/**/*.ts\""
```
Comment (anna-lach) [inline package.json:245]: So maybe better without escaping?

```suggestion
      "command": "eslint --config packages/components/eslint.config.mjs 'packages/components/**/*.ts'"
```

File: website/src/styles/main.scss [hunk 3]
```diff
@@ -4,33 +4,25 @@
...
 @import 'components/tabs';
-
 @import 'categories/design-tokens/base-tokens';
-
 @import 'layout/content';
 @import 'layout/navigation';
-
 @import 'typography';
 @import 'elements';
-
```
Comment (anna-lach) [inline website/src/styles/main.scss:24]: Can we please keep those separators? :) I added them to separate logically differentiated styles. :)

File: website/src/styles/main.scss [hunk 4]
```diff
@@ -4,33 +4,25 @@
...
 @import 'homepage/explore';
 @import 'homepage/call-to-action';
 @import 'homepage/contact-block';
-
 @import 'categories/components';
-
 @import 'basic';
 
-@import 'slds-sanoma-learning/base.css';
+// @import 'slds-sanoma-learning/base.css';
```
Comment (anna-lach) [inline website/src/styles/main.scss:26]: So it looks like it can be removed :)

File: website/src/styles/main.scss (hunk 3 above)
Comment (jpzwarte) [reply website/src/styles/main.scss]: Fixed; `lint-staged` auto removed them.

**Review by anna-lach — APPROVED** [review]

**Comment by Diaan** [comment]
Checklist fails on fonts, even though the fonts are loaded: 
image

Also, the changeset is missing

**Review by Diaan — APPROVED** [review]

**Pull request merged**